app/models/user.rb

```diff
@@ -247,7 +247,7 @@ def confirm_current_password
   end
 
   def new_password_differs_from_current
-    if password == current_password
+    if valid_password?(password)
```

Comment:
How is `valid_password?` the correct test here?

Reply:
As far as I can tell, `valid_password?(password)` tests that the variable `password`, when encrypted with the user's salt, matches the encrypted password in the database. It's a method that's connected to the Devise gem. So calling this method gets around the issue of having to compare a plain text password supplied in the form with the encrypted one in the database. Maybe I misunderstood the issue/goal of this fix and that's not behavior we want to support.